### PR TITLE
Fix error handling in makeDirectoryAsync method on Android. Resolves #3823.

### LINKED ISSUE #3823: makeDirectoryAsync creates error although directories were created properly
## 🐛 Bug Report

First of all i found different discussions about this bug, **but all were closed**. But it is still alive, and it is frustrating for any developer while diving into Expo: makeDirectoryAsync creates an error, however the directories are created.

### Environment
  Expo CLI 2.13.0 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.12.0 - /usr/local/bin/node
      npm: 6.7.0 - /usr/local/bin/npm
    IDEs:
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.6 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1 
    npmGlobalPackages:
      expo-cli: 2.13.0

developing on a **Huawei P10** via Expo App, **Android 8.0**. Expo has permission to read/write in storage.

### Steps to Reproduce

- execute makeDirectoryAsync to create directories like 'foo/bar'. 
- check succes with readDirecoryAsync by reading 'foo'

### Expected Behavior
checking 'foo' via readDirecoryAsync should be like:
DIR Array [ 
    "bar"
 ]

### Actual Behavior
this will result in an error like

ERROR [Error: Directory 'file:///data/user/0/host.exp.exponent/files/ExperienceData/%2540anonymous%252Fhiker-87e39a6d-864b-4d3d-b56a-83f7b51cccd2/foo/bar' could not be created.]

In fact thats nonsens because the directories were created as a check via readDirecoryAsync of 'foo' shows

### Reproducible Demo
Try out that snippet
```
        try {
            await FileSystem.makeDirectoryAsync(FileSystem.documentDirectory + 'foo/bar', { intermediates: true });
        } catch (e) {
            console.info("ERROR", e);
        }
        try {
            const isDir = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory + 'foo');
            console.info("DIR", isDir);
        } catch (e) {
            console.info("ERROR", e);
        }
```
Both outputs are there: the error and the expected success 'DIR'.

### PR DESCRIPTION
**Pull request by sergeichestakov**
This PR changes the behavior of the makeDirectoryAsync method in the FileSystem module on Android to not throw an error when the directory already exists and intermediates is set to true.

This will be consistent with the behavior of the iOS implementation as well as the expected behavior of `mkdir -p` on Unix systems.

# Why

Issue #3823 describes the undesired behavior where makeDirectoryAsync is called on a directory that already exists and an error is thrown when it shouldn't be. 

# How

I added a check to see if the directory already exists in the FileSystem Android module and resolved the promise if that was the case and intermediates was set to true.

# Test Plan

In this commit, I've included a test in the test-suite to make sure that no error is thrown when makeDirectoryAsync is called twice in a row with intermediates set to true. The other tests that call makeDirectoryAsync seem to cover the other edge cases.

### CODE REVIEW

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java [hunk 1]
```diff
@@ -398,10 +398,10 @@ public void makeDirectoryAsync(String uriStr, Map<String, Object> options, Promi
...
-                file.mkdirs() :
-                file.mkdir();
-        if (success) {
+        boolean previouslyCreated = file.isDirectory();
+        boolean setIntermediates = options.containsKey("intermediates") && (Boolean) options.get("intermediates");
+        boolean success = setIntermediates ? file.mkdirs() : file.mkdir();
+        if (success || (setIntermediates && previouslyCreated)) {
           promise.resolve(null);
         } else {
           promise.reject("E_DIRECTORY_NOT_CREATED",
```
Comment (ide) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java]: Update the error message so that it reads `...could not be created or already exists`

**Review by ide — APPROVED** [review]

**Review by ide — APPROVED** [review]
Looks good -> squash and merge your changes (there's a button for it on the PR here) to master.
